bionmr_utils/md/process/calc.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+from typing import Tuple, Dict, List, Callable
+from pyxmolpp2 import calc_autocorr_order_2
+from tqdm import tqdm
+
+
+def calc_autocorr(vectors: Dict[Tuple[int, str], np.ndarray],
+                  calc_autocorr_func: Callable[[np.ndarray], List[float]] = calc_autocorr_order_2,
+                  limit=-1
+                  ) -> Dict[Tuple[str, str], List[float]]:
+    """
+    Get auto-correlation from trajectory
+
+    :param vectors: dict of (rid, aname): VectorXYZ
+    :calc_autocorr_func: function calculates auto-correlation
+    :limit: length of auto-correlation function
+    :return dict of (rid, aname): auto-correlation
+    """
+    autocorr = {(rid, aname): calc_autocorr_func(vector, limit)
+                for (rid, aname), vector in vectors.items()
+                }
+    return autocorr

```

Comment:
avoid creating variables right before return

```suggestion
    return {
        (rid, aname): calc_autocorr_func(vector, limit)
        for (rid, aname), vector in vectors.items()
    }
```